### ISSUE TITLE
Ability to hide a task from doc output

### ISSUE THREAD

**Issue by bachirelkhoury** [rags/pynt#7]
I have a complex chain of tasks that I don't want to be all exposed/available 
when using pynt.

I can use internal python functions instead, but I really like the simplicity of chaining tasks and having consistent logs etc...

Adding @task(hide=True) might be a nice addition

So it looks something like that:
@task(hide=True)
def build():
    '''Build for distribution'''

@task(hide=True)
def upload():
    '''Upload build'''

@task(hide=True)
def install():
    '''Install new build distribution'''

@task()
def restart_server:
    ''Restart the server'''

@task(build, upload, install, restart_server)
def deploy_prod():
    '''Build & deploy app to production server. Then restart server.'''
    print("Deployment Complete!")

At the moment, they are all exposed as tasks, and can appear in pynt output:
Tasks in build file build.py:
  build                       Build for distribution
  clean                      Clean build directory.
  deploy_prod            Build & deploy app to production server. Then restarts server.
  install                     Install new build distribution''
  restart_server          Restart pserve on remote
  upload                    Upload the latest build to server

**Comment by jabbalaci** [comment]
I second this request. I faced the same problem and I had the same idea as a solution: `@task(hide=True)`. In this case the task wouldn't appear in the auto-generated docs and in the output of `pynt -l`. I think it'd be a useful feature.

You suggested putting the private things in a separate file. However, I'd also like to keep things simple and for the build process I only want to manage just one file, `build.py`.

Motivation: say I have 2 main tasks and both need to perform a common sub-task first. If the sub-task is marked with `@task`, then it can be used as a dependency. However, in the docs it shouldn't appear since it is not to be used alone.

**Comment by jabbalaci** [comment]
I really appreciate this project for its simplicity, but if you refuse to add this simple thing (that is needed by more than one users), I'll have to make a fork of the project. Slicing a project is not a good idea, so please consider adding this thing to the project. Thank you!

**Comment by rags** [comment]
Hi Jabba,
First of all thank you the wiki suggestion. I got that done.
I missed your earlier correspondence on this thread. Do you already have an
commit for this? We can get it merged.

If you need me to make those changes I'll need some time. Its a bad time at
work these days. We are in the middle of a system redesign and most of my
time goes into that.
The changes for this is pretty straightforward and will require few hours
of your time. Will you be able to take this up?

**Comment by jabbalaci** [comment]
Hi Rags,

Thank you for not rejecting the idea. I will try to do it, though I'm not very familiar with custom decorators. Is the syntax `@task(hide=True)` OK for you?

L.

**Comment by rags** [comment]
I tested the patch today. It only hides the task from being listed. It can still be run from the commandline. We need to reevaluate what the value add here is.
hide=True may be misleading. list=False might be a better name to indicate the task is simply not listed

**Comment by jabbalaci** [comment]
As for me, this is what I wanted. Hide tasks in the auto-generated docs. pynt is a simple build tool, there's no need to complicate things. If you prefer list=False, that's also fine. I simply don't want to see some sub-tasks in the list.

@bachirelkhoury : what do you think? You also had this feature request.

**Comment by rags** [comment]
The more I think about it the more I'm leaning towards having tasks with leading underscore as hidden by convention (python convention)

so
```lang=python

#will be hidden/private task
@task()
def _foo(): print("implementation details")

#This will be a listed task
@task(_foo)
def foo(): print("interface")

```
The pynt -l will only list foo and not _foo (which is a private dependency)

**Comment by rags** [comment]
having underscore prefix convention is intuitive and it doesnt require extra attributes like hide, show, list

**Comment by jabbalaci** [comment]
A leading underscore is also a good idea.

**Comment by jabbalaci** [comment]
How to proceed? Do you want me to implement the hiding by underscore?

**Comment by rags** [comment]
Yes, I have merged your changes. But I think underscore solution makes
sense. It only changes the way we determine the 'hide' boolean you added.
Its busy time for me at work. This will take 1/2 a day or so to implement
update documentation, examples and tests

**Comment by jabbalaci** [comment]
PR sent: https://github.com/rags/pynt/pull/17 . I only updated the source code, not the documentation. I go on holiday tomorrow for 5 days, so I can help with the docs / tests later.

**Comment by rags** [comment]
This feature has been released as part of pynt 0.8.2. Use "pip install --upgrade pynt" to get this feature.

**Comment by jabbalaci** [comment]
Thank you for integrating this feature! And thanks for updating the docs and tests.
